test/distributed/test_c10d_nccl.py

```diff
@@ -3172,11 +3172,16 @@ def test_nccl_user_buffer_registration(self):
     @requires_multicast_support()
     def test_nccl_window_registration(self):
         store = c10d.FileStore(self.file_name, self.world_size)
+        torch.cuda.set_device(self.rank)
```

Comment:
Ugh, should we need put this in a with context and return the default cuda device to original rank? Otherwise order of ops could matter right?

Reply:
yes, should put it in context manager